python/pyiceberg/transforms.py

```diff
@@ -312,6 +293,28 @@ def source_type(self) -> IcebergType:
     def width(self) -> int:
         return self._width
 
+    def hash_function(self, source: IcebergType) -> Callable[[Optional[S]], Optional[S]]:
+        source_type = type(source)
+        if source_type in {IntegerType, LongType}:
+
+            def truncate_func(v):
+                return v - v % self._width
+
+        elif source_type in {StringType, BinaryType}:
+
+            def truncate_func(v):
+                return v[0 : min(self._width, len(v))]
```

Comment:
Nice that this reduces code duplication as well.

Reply:
Yes, like it a lot